src/repr/src/scalar.rs

```diff
@@ -1745,6 +1753,93 @@ impl<'a> ScalarType {
     }
 }
 
+#[cfg(test)]
```

Comment:
Urgh... I was hoping we don't need to resort to custom implementations. Can you reference a guide or something that you used to implement this?